fvm/scriptEnv.go

```diff
@@ -0,0 +1,750 @@
+package fvm
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/onflow/cadence"
+	jsoncdc "github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/opentracing/opentracing-go"
+	traceLog "github.com/opentracing/opentracing-go/log"
+
+	"github.com/onflow/flow-go/fvm/blueprints"
+	"github.com/onflow/flow-go/fvm/crypto"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/handler"
+	"github.com/onflow/flow-go/fvm/programs"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/fvm/utils"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/trace"
+	"github.com/onflow/flow-go/storage"
+)
+
+var _ runtime.Interface = &ScriptEnv{}
+
+// ScriptEnv is a read-only mostly used for executing scripts.
+type ScriptEnv struct {
+	ctx           Context
+	sth           *state.StateHolder
+	vm            *VirtualMachine
+	accounts      *state.Accounts
+	contracts     *handler.ContractHandler
+	programs      *handler.ProgramsHandler
+	accountKeys   *handler.AccountKeyHandler
+	metrics       *handler.MetricsHandler
+	uuidGenerator *state.UUIDGenerator
+	logs          []string
+	totalGasUsed  uint64
+	rng           *rand.Rand
+	traceSpan     opentracing.Span
+}
+
+func NewScriptEnvironment(ctx Context, vm *VirtualMachine, sth *state.StateHolder, programs *programs.Programs) *ScriptEnv {
```

Comment:
This looks like a copy of `TransactionEnvironment`. Maybe we should have a one common base for both?

Reply:
it looks the same now, but won't be the same soon, when we switch the handlers with readonly version of them.